src/Sulu/Bundle/AdminBundle/Tests/Unit/Metadata/SchemaMetadata/PropertyMetadataMapper/BlockPropertyMetadataMapperTest.php

```diff
@@ -0,0 +1,206 @@
+        ], $result->toJsonSchema());
+    }
+
+    public function testMapPropertyMetatadaWithRequiredFalse(): void
+    {
+        $types = [
+            'headline' => [
+                'isGlobalBlock' => true,
+            ],
+            'text' => [
+                'children' => [
+                    'text' => 'text_line',
+                ],
+            ],
+        ];
+
+        $metadata = new FieldMetadata('imageMap');
+        $metadata->setRequired(false);
+        foreach ($types as $key => $config) {
+            $type = new FormMetadata();
+            $type->setKey($key);
+
+            $isGlobalBlock = $config['isGlobalBlock'] ?? false;
+            if ($isGlobalBlock) {
+                $tagMetadata = new TagMetadata();
+                $tagMetadata->setName('sulu.global_block');
+                $tagMetadata->setAttributes([
+                    'global_block' => $key,
+                ]);
+                $type->addTag($tagMetadata);
+            }
+
+            foreach ($config['children'] ?? [] as $childName => $childType) {
+                $itemMetadata = new FieldMetadata($childName);
+                $type->addItem($itemMetadata);
+            }
+
+            if (!$isGlobalBlock) {
+                $itemSchemaMetadata = new SchemaMetadata([
+                    new SchemaPropertyMetadata('type', false),
+                ]);
+                $this->schemaMetadataProvider->getMetadata($type->getItems())->willReturn($itemSchemaMetadata);
+            }
+            $metadata->addType($type);
+        }
+
+        $result = $this->blockPropertyMetadataMapper->mapPropertyMetadata($metadata);
+        $this->assertSame([
+            'type' => 'array',
+            'items' => [
+                'allOf' => [
+                    [
+                        'if' => [
+                            'type' => 'object',
+                            'properties' => [
+                                'type' => [
+                                    'const' => 'headline',
+                                ],
+                            ],
+                            'required' => ['type'],
+                        ],
+                        'then' => [
+                            '$ref' => '#/definitions/headline',
+                        ],
+                    ],
+                    [
+                        'if' => [
+                            'type' => 'object',
+                            'properties' => [
+                                'type' => [
+                                    'const' => 'text',
+                                ],
+                            ],
+                            'required' => ['type'],
+                        ],
+                        'then' => [
+                            'type' => ['number', 'string', 'boolean', 'object', 'array', 'null'],
+                        ],
+                    ],
+                ],
+            ],
+        ], $result->toJsonSchema());
+    }
```

Comment:
The block didn't have any tests before I copied it from the image map test case and adopted it.